### PR TITLE
[1.11] Added Tesla Energy support. Closes #100

### LINKED ISSUE #100: Forge Energy/Tesla energy support
TOP has native RF support, might be nice to also have Tesla and Forge energy support.

### PR DESCRIPTION
**Pull request by tomelfring**
And why not, added also Tesla support.

Added Tesla support. Also adds an optional dependency.
Does work without Tesla.

### CODE REVIEW

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java [hunk 1]
```diff
@@ -188,10 +190,15 @@ private void showRF(IProbeInfo probeInfo, World world, BlockPos pos) {
             if (handler != null) {
                 addRFInfo(probeInfo, config, handler.getEnergyStored(), handler.getMaxEnergyStored());
             }
+        } else if (Loader.isModLoaded("tesla") && te != null && te.hasCapability(TeslaCapabilities.CAPABILITY_HOLDER, null)) {
```
Comment (McJty) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:193]: Will the access to TeslaCapabilities work properly if testla is not present?

Comment (tomelfring) [reply src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:193]: It won't come to that statement if Tesla isn't loaded.
Loader.isModLoaded("tesla") will return false if it's not loaded, which results in skipping the rest of the if statement.

Comment (McJty) [reply src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:193]: Ok, and you have actually tested that? I would feel safer to include this if that was properly tested

Comment (tomelfring) [reply src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:193]: I've tested it. I'll post some screenshots in a couple minutes (starting everything)

Comment (tomelfring) [reply src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:193]: Used a WIP mod of mine, disabled the ForgeEnergy support, leaving only Tesla.
http://imgur.com/a/akKOz

Confirmed that the first case does not crash, no errors in console whatsoever. Second case it correctly find the build-in energy capabity with Tesla (confirmed with logging).

**Comment by McJty** [comment]
I had to revert tesla support for now because I cannot get it to work in my dev env. setupDecompWorkspace seems to run without errors but Intellij cannot find the needed Tesla classes so DefaultProbeInfoProvider doesn't compile. Can you look into fixing this? Thanks

**Pull request merged**